examples/vision/semantic_image_clustering.py

```diff
@@ -0,0 +1,576 @@
+"""
+Title: Semantic Image Clustering
+Author: [Khalid Salama](https://www.linkedin.com/in/khalid-salama-24403144/)
+Date created: 2021/02/28
+Last modified: 2021/02/28
+Description: Semantic Clustering by Adopting Nearest neighbors (SCAN) algorithm.
+"""
+
+"""
+## Introduction
+
+This example demonstrates applying the [Semantic Clustering by Adopting Nearest neighbors
+(SCAN)](https://arxiv.org/abs/2005.12320) algorithm, by Wouter Van Gansbeke et. al, on the
+[CIFAR-10](https://www.cs.toronto.edu/~kriz/cifar.html) images dataset. The algorithm,
+consists of two phases:
+1. Self-supervised visual representation learning of images, in which we use the
+[simCLR](https://arxiv.org/abs/2002.05709?ref=hackernoon.com) technique.
+2. Clustering of the learnt visual representation vectors that maximizes the agreement
+between the cluster assignments of the neighbours.
```

Comment:
I think the style guides suggest we use US English:

```suggestion
between the cluster assignments of the neighbors.
```